src/jit/emitxarch.cpp

```diff
@@ -11196,9 +11196,21 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
     assert(sz == emitSizeOfInsDsc(id));
 
 #if !FEATURE_FIXED_OUT_ARGS
+    bool bUpdateStackLv = true;
+
+    if (emitIGisInProlog(ig))
+        bUpdateStackLv = false;
+    if (emitIGisInEpilog(ig))
+        bUpdateStackLv = false;
+#if FEATURE_EH_FUNCLETS
+    if (emitIGisInFuncletProlog(ig))
+        bUpdateStackLv = false;
+    if (emitIGisInFuncletEpilog(ig))
+        bUpdateStackLv = false;
+#endif // FEATURE_EH_FUNCLETS
```

Comment:
@janvorli I think you mean to use `FEATURE_EH_FUNCLETS && defined(_TARGET_X86_)` to keep the behavior unchanged for Windows x64. I agree with this. It's possible (probable even?) that this wouldn't change x64 behavior, but it might be better to be safe.

IMO, the variable name should be `updateStackLevel` (don't use Hungarian, don't abbreviate). I like @janvorli's suggestion, except I don't like bitwise operators on bool, so I suggest:
```C++
bool updateStackLevel =!emitIGisInProlog(ig) && !emitIGisInEpilog(ig);
#if FEATURE_EH_FUNCLETS && defined(_TARGET_X86_) // This might also be applicable to x64.
updateStackLevel = updateStackLevel && !emitIGisInFuncletProlog(ig) && !emitIGisInFuncletEpilog(ig);
#endif // FEATURE_EH_FUNCLETS && defined(_TARGET_X86_)
```

---
In reply to: [96857183](https://github.com/dotnet/coreclr/pull/9001#discussion_r96857183) [](ancestors = 96857183)